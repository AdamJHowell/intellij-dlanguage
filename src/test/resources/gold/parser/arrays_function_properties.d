int[] array;
void foo(int[] a, int x);

foo(array, 3);
array.foo(3);   // means the same thing