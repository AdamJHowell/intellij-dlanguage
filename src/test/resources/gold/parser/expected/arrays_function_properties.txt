D Language File
  DLanguageDeclarationImpl(DECLARATION)
    DLanguageTypeImpl(TYPE)
      DLanguageType_2Impl(TYPE_2)
        DLanguageBuiltinTypeImpl(BUILTIN_TYPE)
          PsiElement(DLanguageTokenType.int)('int')
      DLanguageTypeSuffixImpl(TYPE_SUFFIX)
        PsiElement(DLanguageTokenType.[)('[')
        PsiElement(DLanguageTokenType.])(']')
    DLanguageVariableDeclarationImpl(VARIABLE_DECLARATION)
      DECLARATOR
        IDENTIFIER
          PsiElement(DLanguageTokenType.ID)('array')
      PsiElement(DLanguageTokenType.;)(';')
  DLanguageDeclarationImpl(DECLARATION)
    DLanguageTypeImpl(TYPE)
      DLanguageType_2Impl(TYPE_2)
        DLanguageBuiltinTypeImpl(BUILTIN_TYPE)
          PsiElement(DLanguageTokenType.void)('void')
    FUNCTION_DECLARATION
      IDENTIFIER
        PsiElement(DLanguageTokenType.ID)('foo')
      DLanguageParametersImpl(PARAMETERS)
        PsiElement(DLanguageTokenType.()('(')
        PARAMETER
          DLanguageTypeImpl(TYPE)
            DLanguageType_2Impl(TYPE_2)
              DLanguageBuiltinTypeImpl(BUILTIN_TYPE)
                PsiElement(DLanguageTokenType.int)('int')
            DLanguageTypeSuffixImpl(TYPE_SUFFIX)
              PsiElement(DLanguageTokenType.[)('[')
              PsiElement(DLanguageTokenType.])(']')
          IDENTIFIER
            PsiElement(DLanguageTokenType.ID)('a')
        PsiElement(DLanguageTokenType.,)(',')
        PARAMETER
          DLanguageTypeImpl(TYPE)
            DLanguageType_2Impl(TYPE_2)
              DLanguageBuiltinTypeImpl(BUILTIN_TYPE)
                PsiElement(DLanguageTokenType.int)('int')
          IDENTIFIER
            PsiElement(DLanguageTokenType.ID)('x')
        PsiElement(DLanguageTokenType.))(')')
      PsiElement(DLanguageTokenType.;)(';')
  DLanguageDeclarationImpl(DECLARATION)
    UNITTEST
      UNITTEST
        PsiElement(DLanguageTokenType.unittest)('unittest')
        DLanguageBlockStatementImpl(BLOCK_STATEMENT)
          PsiElement(DLanguageTokenType.{)('{')
          DLanguageDeclarationsAndStatementsImpl(DECLARATIONS_AND_STATEMENTS)
            DLanguageDeclarationOrStatementImpl(DECLARATION_OR_STATEMENT)
              DLanguageStatementImpl(STATEMENT)
                DLanguageStatementNoCaseNoDefaultImpl(STATEMENT_NO_CASE_NO_DEFAULT)
                  DLanguageExpressionStatementImpl(EXPRESSION_STATEMENT)
                    DLanguageExpressionImpl(EXPRESSION)
                      DLanguageAssignExpressionImpl(ASSIGN_EXPRESSION)
                        DLanguageCmpExpressionImpl(CMP_EXPRESSION)
                          DLanguageUnaryExpressionImpl(UNARY_EXPRESSION)
                            DLanguageFunctionCallExpressionImpl(FUNCTION_CALL_EXPRESSION)
                              DLanguageUnaryExpressionImpl(UNARY_EXPRESSION)
                                DLanguagePrimaryExpressionImpl(PRIMARY_EXPRESSION)
                                  DLanguageIdentifierOrTemplateInstanceImpl(IDENTIFIER_OR_TEMPLATE_INSTANCE)
                                    IDENTIFIER
                                      PsiElement(DLanguageTokenType.ID)('foo')
                              DLanguageArgumentsImpl(ARGUMENTS)
                                PsiElement(DLanguageTokenType.()('(')
                                DLanguageArgumentListImpl(ARGUMENT_LIST)
                                  DLanguageAssignExpressionImpl(ASSIGN_EXPRESSION)
                                    DLanguageCmpExpressionImpl(CMP_EXPRESSION)
                                      DLanguageUnaryExpressionImpl(UNARY_EXPRESSION)
                                        DLanguagePrimaryExpressionImpl(PRIMARY_EXPRESSION)
                                          DLanguageIdentifierOrTemplateInstanceImpl(IDENTIFIER_OR_TEMPLATE_INSTANCE)
                                            IDENTIFIER
                                              PsiElement(DLanguageTokenType.ID)('array')
                                  PsiElement(DLanguageTokenType.,)(',')
                                  DLanguageAssignExpressionImpl(ASSIGN_EXPRESSION)
                                    DLanguageCmpExpressionImpl(CMP_EXPRESSION)
                                      DLanguageUnaryExpressionImpl(UNARY_EXPRESSION)
                                        DLanguagePrimaryExpressionImpl(PRIMARY_EXPRESSION)
                                          PsiElement(DLanguageTokenType.INTEGER_LITERAL)('3')
                                PsiElement(DLanguageTokenType.))(')')
                    PsiElement(DLanguageTokenType.;)(';')
            DLanguageDeclarationOrStatementImpl(DECLARATION_OR_STATEMENT)
              DLanguageStatementImpl(STATEMENT)
                DLanguageStatementNoCaseNoDefaultImpl(STATEMENT_NO_CASE_NO_DEFAULT)
                  DLanguageExpressionStatementImpl(EXPRESSION_STATEMENT)
                    DLanguageExpressionImpl(EXPRESSION)
                      DLanguageAssignExpressionImpl(ASSIGN_EXPRESSION)
                        DLanguageCmpExpressionImpl(CMP_EXPRESSION)
                          DLanguageUnaryExpressionImpl(UNARY_EXPRESSION)
                            DLanguageFunctionCallExpressionImpl(FUNCTION_CALL_EXPRESSION)
                              DLanguageUnaryExpressionImpl(UNARY_EXPRESSION)
                                DLanguagePrimaryExpressionImpl(PRIMARY_EXPRESSION)
                                  DLanguageIdentifierOrTemplateInstanceImpl(IDENTIFIER_OR_TEMPLATE_INSTANCE)
                                    IDENTIFIER
                                      PsiElement(DLanguageTokenType.ID)('array')
                                PsiElement(DLanguageTokenType..)('.')
                                DLanguageIdentifierOrTemplateInstanceImpl(IDENTIFIER_OR_TEMPLATE_INSTANCE)
                                  IDENTIFIER
                                    PsiElement(DLanguageTokenType.ID)('foo')
                              DLanguageArgumentsImpl(ARGUMENTS)
                                PsiElement(DLanguageTokenType.()('(')
                                DLanguageArgumentListImpl(ARGUMENT_LIST)
                                  DLanguageAssignExpressionImpl(ASSIGN_EXPRESSION)
                                    DLanguageCmpExpressionImpl(CMP_EXPRESSION)
                                      DLanguageUnaryExpressionImpl(UNARY_EXPRESSION)
                                        DLanguagePrimaryExpressionImpl(PRIMARY_EXPRESSION)
                                          PsiElement(DLanguageTokenType.INTEGER_LITERAL)('3')
                                PsiElement(DLanguageTokenType.))(')')
                    PsiElement(DLanguageTokenType.;)(';')
          PsiComment(DLanguageTokenType.LINE_COMMENT)('// means the same thing')
          PsiElement(DLanguageTokenType.})('}')